ci/ci/ci.py

```diff
@@ -113,12 +117,47 @@ async def dev_deploy_branch(request, userdata):
     except Exception as e:  # pylint: disable=broad-except
         message = traceback.format_exc()
         log.info('dev deploy failed: ' + message, exc_info=True)
-        raise web.HTTPBadRequest(text=f'starting the deploy failed due to\n{message}') from e
+        raise web.HTTPBadRequest(
+            text=f'starting the deploy failed due to\n{message}'
+        ) from e
     return web.json_response({'sha': sha, 'batch_id': batch_id})
 
 
+# This is CPG-specific, as the Hail team redeploys by watching the main branch.
+@routes.post('/api/v1alpha/prod_deploy')
+@rest_authenticated_developers_only
+async def prod_deploy(request, unused_userdata):
+    """Deploys the main branch to the production namespace ("default")."""
+
+    app = request.app
+    try:
+        params = await request.json()
+    except Exception as e:
+        message = 'could not read body as JSON'
+        log.info('prod deploy failed: ' + message, exc_info=True)
+        raise web.HTTPBadRequest(text=message) from e
+
+    try:
+        steps = params['steps']
+    except Exception as e:
+        message = f'parameters are wrong; check the steps syntax.\n\n{params}'
+        log.info('prod deploy failed: ' + message, exc_info=True)
+        raise web.HTTPBadRequest(text=message) from e
+
+    watched_branch = WatchedBranch(
+        0, FQBranch.from_short_str('populationgenomics/hail:main'), True
```

Comment:
Should we make this value configurable, it might be useful upstream in the future.

Reply:
This is something we're not going to upstream (see the comment above the function), as upstream Hail uses a very custom and sophisticated mechanism to authorize PRs and then update the deployment automatically. It's somewhat nicer in the sense that it's actual CI, whereas our process is obviously manual. But it's also way more complicated (and I think it's nice to be able to time deployments and watch that nothing breaks).

Reply:
I still think it would be good to pull the `populationgenomics/hail` value from a configuration parameter like `repository` or something. The less CPG hard-coded specific CGP constants the better IMO.
But I've approved the PR, and still happy to for you to merge :)

Reply:
Summarizing our separate chat: in general I completely agree with you -- in this case it's a CPG-specific hack that wouldn't benefit from configurability like other code that we'd want to upstream.